### PR TITLE
(DE-652) feat(analytics): custom schemas for local development

### PR DESCRIPTION
**Pull request by valoumiaou**
# DE PR

## Describe your changes

Please include a summary of the changes:

This PR [adds/removes/fixes/replaces] the [feature/bug/etc]. 

Tag a reviewer if necessacy  @github/username 

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/dbt_project.yml [hunk 1]
```diff
@@ -28,6 +28,22 @@ clean-targets:
 
 models:
   data_gcp_dbt:
+    +pre-hook:
+      - "{% if target.name == 'local' %} \
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/dbt_project.yml:32]: On pourrait mettre cette macro dans un fichier séparé pour pas surcharger le yaml

File: orchestration/dags/data_gcp_dbt/dbt_project.yml [hunk 2]
```diff
@@ -28,6 +28,22 @@ clean-targets:
...
+          CREATE SCHEMA IF NOT EXISTS `{{ target.project }}.tmp_{{ env_var('USER', 'anonymous_user') }}`; \
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/dbt_project.yml:33]: où est ce que la env_var user est définie ?

File: orchestration/dags/data_gcp_dbt/dbt_project.yml [hunk 3]
```diff
@@ -28,6 +28,22 @@ clean-targets:
...
+          ALTER SCHEMA `{{ target.project }}.tmp_{{ env_var('USER', 'anonymous_user') }}` \
+          SET OPTIONS( \
+            default_table_expiration_days = 2 \
+          ); \
+        {% endif %}"
+    +post-hook:
+      - "{% if target.name == 'local' and config.get('materialized') == 'view' %} \
+          ALTER VIEW {{ this }} \
+          SET OPTIONS(expiration_timestamp = TIMESTAMP_SECONDS(UNIX_SECONDS(CURRENT_TIMESTAMP()) + 172800)); \
+        {% elif target.name == 'local'%} \
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/dbt_project.yml:43]: On pourrait pas tout mettre en view quand on dev en local ?

File: orchestration/dags/data_gcp_dbt/macros/generate_schema_name.sql [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+    {%- set user_name = env_var('USER', 'anonymous_user') -%}
     {%- set is_orchestrated = target.name in ["prod", "stg", "dev"] and target.profile_name != "sandbox" -%}
     {%- set is_elementary = target.profile_name == "elementary" -%}
 
-    {%- if target.profile_name == "CI" or target.name == "local" or is_elementary -%}
+    {%- if target.profile_name == "CI" or is_elementary -%}
         {{ default_schema }}
 
+     {%- elif target.name == "local" -%}
+        {{ "tmp_" ~ user_name }}
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/macros/generate_schema_name.sql:12]: ajouter _dev en suffixe ?

File: orchestration/dags/data_gcp_dbt/dbt_project.yml (hunk 2 above)
Comment (valoumiaou) [reply orchestration/dags/data_gcp_dbt/dbt_project.yml:33]: variable d'environnement defini sur tte les machine

**Review by LucileRainteau — APPROVED** [review]

**Pull request merged**